lib/Doctrine/Migrations/DependencyFactory.php

```diff
@@ -63,141 +101,169 @@ public function getSchemaDumper() : SchemaDumper
         });
     }
 
+    private function getSchemaProvider() : SchemaProviderInterface
+    {
+        // @todo what about the other schema providers?
+        return $this->getDependency(SchemaProviderInterface::class, function () : SchemaProviderInterface {
+            return new OrmSchemaProvider($this->em);
+        });
+    }
+
+    public function getDiffGenerator() : DiffGenerator
+    {
+        return $this->getDependency(DiffGenerator::class, function () : DiffGenerator {
+            return new DiffGenerator(
+                $this->getConnection()->getConfiguration(),
+                $this->getConnection()->getSchemaManager(),
+                $this->getSchemaProvider(),
+                $this->getConnection()->getDatabasePlatform(),
+                $this->getMigrationGenerator(),
+                $this->getMigrationSqlGenerator()
+            );
+        });
+    }
+
     public function getSchemaDiffProvider() : SchemaDiffProviderInterface
     {
         return $this->getDependency(SchemaDiffProviderInterface::class, function () : LazySchemaDiffProvider {
             return LazySchemaDiffProvider::fromDefaultProxyFactoryConfiguration(
                 new SchemaDiffProvider(
-                    $this->getConnection()->getSchemaManager(),
-                    $this->getConnection()->getDatabasePlatform()
+                    $this->connection->getSchemaManager(),
+                    $this->connection->getDatabasePlatform()
                 )
             );
         });
     }
 
     public function getFileBuilder() : FileBuilder
     {
-        return $this->getDependency(FileBuilder::class, function () : FileBuilder {
-            return new FileBuilder(
-                $this->getConnection()->getDatabasePlatform(),
-                $this->configuration->getMigrationsTableName(),
-                $this->configuration->getQuotedMigrationsColumnName(),
-                $this->configuration->getQuotedMigrationsExecutedAtColumnName()
-            );
+        return $this->getDependency(FileBuilder::class, static function () : FileBuilder {
+            return new FileBuilder();
         });
     }
 
     public function getParameterFormatter() : ParameterFormatterInterface
     {
         return $this->getDependency(ParameterFormatter::class, function () : ParameterFormatter {
-            return new ParameterFormatter($this->getConnection());
+            return new ParameterFormatter($this->connection);
         });
     }
 
+    public function getMigrationsFinder() : MigrationFinder
+    {
+        $finder =  $this->getDependency(GlobFinder::class, static function () : MigrationFinder {
+            return new GlobFinder();
+        });
+
+        // todo move this to DI
```

Comment:
can it be done ?

Reply:
The ideas is that based on the migrations folder schema, here should be passed the right finder